cmd/calyptia/get_cluster_objects.go

```diff
@@ -0,0 +1,224 @@
+	if err != nil {
+		return nil, fmt.Errorf("could not prefetch core-instances: %w", err)
+	}
+
+	if len(aa.Items) == 0 {
+		return nil, nil
+	}
+
+	var clusterobjects []cloud.ClusterObject
+	var mu sync.Mutex
+
+	g, gctx := errgroup.WithContext(config.ctx)
+	for _, a := range aa.Items {
+		a := a
+		g.Go(func() error {
+			objects, err := config.cloud.ClusterObjects(gctx, a.ID,
+				cloud.ClusterObjectParams{})
+			if err != nil {
+				return err
+			}
+
+			mu.Lock()
+			clusterobjects = append(clusterobjects, objects.Items...)
+			mu.Unlock()
+
+			return nil
+		})
+	}
+
+	if err := g.Wait(); err != nil {
+		return nil, err
+	}
+
+	var uniqueClusterObjects []cloud.ClusterObject
+	clusterObjectIDs := map[string]struct{}{}
+	for _, coid := range clusterobjects {
+		if _, ok := clusterObjectIDs[coid.ID]; !ok {
+			uniqueClusterObjects = append(uniqueClusterObjects, coid)
+			clusterObjectIDs[coid.ID] = struct{}{}
+		}
+	}
+
+	return uniqueClusterObjects, nil
+}
+
+func (config *config) completeClusterObjects(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	pp, err := config.fetchAllClusterObjects()
+	if err != nil {
+		cobra.CompError(err.Error())
+		return nil, cobra.ShellCompDirectiveError
+	}
+
+	if pp == nil {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+
+	return clusterObjectsKeys(pp), cobra.ShellCompDirectiveNoFileComp
+}
+
+// ClusterObjectsKeys returns unique cluster object names first and then IDs.
+func clusterObjectsKeys(aa []cloud.ClusterObject) []string {
+	namesCount := map[string]int{}
+	for _, a := range aa {
+		if _, ok := namesCount[a.Name]; ok {
+			namesCount[a.Name] += 1
+			continue
+		}
+
+		namesCount[a.Name] = 1
+	}
+
+	var out []string
+
+	for _, a := range aa {
+		var nameIsUnique bool
+		for name, count := range namesCount {
+			if a.Name == name && count == 1 {
+				nameIsUnique = true
+				break
+			}
+		}
+		if nameIsUnique {
+			out = append(out, a.Name)
+			continue
+		}
+
+		out = append(out, a.ID)
+	}
+
+	return out
+}
+
+// ClusterObjectsUnique returns unique cluster object names.
+func clusterObjectsUniqueByName(aa []cloud.ClusterObject) []cloud.ClusterObject {
+	namesCount := map[string]int{}
+	for _, a := range aa {
+		if _, ok := namesCount[a.Name]; !ok {
+			namesCount[a.Name] = 0
+		}
+		namesCount[a.Name]++
+	}
+
+	var out []cloud.ClusterObject
+	for _, a := range aa {
+		for name, count := range namesCount {
+			if a.Name == name && count == 1 {
+				out = append(out, a)
+				break
+			}
+		}
+	}
+	return out
+}
+
+func (config *config) loadClusterObjectID(key string, environmentID string) (string, error) {
+	aa, err := config.fetchAllClusterObjects()
+	if err != nil {
+		return "", err
+	}
+
+	objs := clusterObjectsUniqueByName(aa)
+	for _, obj := range objs {
+		if obj.Name == key {
+			return obj.ID, nil
+		}
+	}
+
+	return "", fmt.Errorf("unable to find unique key")
```

Comment:
Still this method is not returning correctly. I think.

Reply:
Like, it's getting all cluster objects from all core instances. Then filters all of them that have a unique name, then gets the one that matches the name.
This función then only works by passing a name. But doesn't work if you pass an ID.

Reply:
I forgot to add a call to validUUID to return it as an ID. Should I check that before or after checking it as a name?